org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMDocument.java

```diff
@@ -307,6 +307,24 @@ public boolean hasProlog() {
 		return (children != null && !children.isEmpty() && children.get(0).isProlog());
 	}
 
+	/**
+	 * Returns true if this document has a DTD DOCTYPE
+	 * declaration. Returns false otherwise.
+	 * 
+	 * @return true if this document has a DTD DOCTYPE
+	 * declaration. Returns false otherwise.
+	 */
+	public boolean hasDoctype() {

```

Comment:
I think you don"t need this method. Please remove it and use DOMDocumentType docType = document.getDoctype();.